bin/proving-service/src/commands/proxy.rs

```diff
@@ -88,19 +105,28 @@ impl StartProxy {
         http_server_options.h2c = true;
         logic.server_options = Some(http_server_options);
 
-        // Enable Prometheus metrics
-        let mut prometheus_service_http =
-            pingora::services::listening::Service::prometheus_http_service();
-        prometheus_service_http.add_tcp(
-            format!(
-                "{}:{}",
+        // Enable Prometheus metrics if enabled in the configuration
+        if self.proxy_config.metrics_config.enable_metrics {
+            let mut prometheus_service_http =
+                pingora::services::listening::Service::prometheus_http_service();
+            prometheus_service_http.add_tcp(
+                format!(
+                    "{}:{}",
+                    self.proxy_config.metrics_config.prometheus_host,
+                    self.proxy_config.metrics_config.prometheus_port
+                )
+                .as_str(),
+            );
+            server.add_service(prometheus_service_http);
+            tracing::info!(
+                "Prometheus metrics enabled on {}:{}",
```

Comment:
nit: To avoid ambiguities (like the question Mirko had) maybe this could mention that metrics are being served from there